src/Chronos.php

```diff
@@ -736,13 +736,15 @@ public static function createFromArray(array $values): static
     /**
      * Create an instance from a timestamp
      *
-     * @param int $timestamp The timestamp to create an instance from.
+     * @param float|int $timestamp The timestamp to create an instance from.
      * @param \DateTimeZone|string|null $timezone The DateTimeZone object or timezone name the new instance should use.
      * @return static
      */
-    public static function createFromTimestamp(int $timestamp, DateTimeZone|string|null $timezone = null): static
+    public static function createFromTimestamp(float|int $timestamp, DateTimeZone|string|null $timezone = null): static
     {
-        return static::now($timezone)->setTimestamp($timestamp);
+        $instance = PHP_VERSION_ID >= 80400 ? parent::createFromTimestamp($timestamp) : new static('@' . $timestamp);
```

Comment:
Now that we accept float, do we need to do any special rounding/etc to mimic the 8.4 behavior when converting to a @timestamp ?

I hope not. I assume they expect the timestamp to still be an integer in float format.

Reply:
> I assume they expect the timestamp to still be an integer in float format.

No, the fractional part of floats ie used as microseconds https://3v4l.org/OT84B#v8.1.0

Reply:
ah, ok. Wasn't sure if the @ format would throw an error, but I think that's ok if it does.